openid-federation-1_0.xml

```diff
@@ -6050,12 +6050,38 @@ redirect_uri=https%3A%2F%2Frp.example.com%2Fauthz_cb
           </section>
         </section>
 
+	<section title="Successful Authentication Response" anchor="AuthenticationResponse">
+	  <t>
+	    The response to a successful authentication request
+	    when using Automatic Registration is the same as the
+	    successful authentication responses defined in <xref target="OpenID.Core"/>.
+	    In particular, it is a successful OAuth 2.0 authorization response
+	    sent via redirection to the Client's redirection URI.
+	  </t>
+	</section>
+
         <section title="Authentication Error Response" anchor="AuthenticationError">
           <t>
-            If the OP fails to establish trust with the RP, it SHOULD use an
-            appropriate error code and an
-            <spanx style="verb">error_description</spanx>
-            that aids the RP in understanding what is wrong.
+	    The error response to an unsuccessful authentication request
+	    when using Automatic Registration is the same as the
+	    error authentication responses defined in <xref target="OpenID.Core"/>.
+	    In particular, it is an OAuth 2.0 authorization error response
+	    sent via redirection to the Client's redirection URI.
+	  </t>
+	  <t>
+	    However, as in both <xref target="OpenID.Core"/> and <xref target="RFC6749"/>,
+	    if the redirection URI is invalid, redirection MUST NOT be performed,
+	    and instead the Authorization Server SHOULD inform the End-User
+	    of the error in the user interface.
+	    The Authorization Server MAY also choose to do this if it
+	    has reason to believe that the redirection URI might be being used
+	    as a component of an open redirector.
+	  </t>
+	  <t>
+            If the OP fails to establish trust with the RP, it SHOULD use
+            appropriate <spanx style="verb">error</spanx> code and
+            <spanx style="verb">error_description</spanx> values
+            to enable the RP to understand what went wrong.
```

Comment:
It would also help to explain in what contexts the `missing_trust_anchor` and `validation_failed` errors apply and can be passed back to the caller, because when the trust establishment fails the `redirect_uri` must not be used. From this follows that these codes apply to PAR with automatic registration only. They are also useful in explicit registration requests, but currently are not defined for use there.

Reply:
[This commit](https://github.com/openid/federation/pull/82/commits/04a8775724421d3e98700fda7eff2eb36a9d371c) says that redirection should not be performed when trust cannot be established.  It also says that the new error codes can be used with PAR.